File: api/src/main/java/org/pmiops/workbench/db/dao/DataDictionaryEntryDao.java
package org.pmiops.workbench.db.dao;

import java.util.List;
import java.util.Optional;
import org.pmiops.workbench.db.model.DbCdrVersion;
import org.pmiops.workbench.db.model.DbDataDictionaryEntry;
import org.springframework.data.repository.CrudRepository;

public interface DataDictionaryEntryDao extends CrudRepository<DbDataDictionaryEntry, Long> {

  Optional<DbDataDictionaryEntry> findByRelevantOmopTableAndFieldNameAndCdrVersion(
      String relevantOmopTable, String fieldName, DbCdrVersion cdrVersion);

  List<DbDataDictionaryEntry> findByFieldNameAndCdrVersion(
      String fieldName, DbCdrVersion cdrVersion);
}

File: api/src/main/java/org/pmiops/workbench/db/dao/DataDictionaryEntryDao.kt
package org.pmiops.workbench.db.dao

import java.util.Optional
import org.pmiops.workbench.db.model.CdrVersion
import org.pmiops.workbench.db.model.DataDictionaryEntry
import org.springframework.data.repository.CrudRepository

interface DataDictionaryEntryDao : CrudRepository<DataDictionaryEntry, Long> {

    fun findByRelevantOmopTableAndFieldNameAndCdrVersion(
            relevantOmopTable: String, fieldName: String, cdrVersion: CdrVersion): Optional<DataDictionaryEntry>

    fun findByFieldNameAndCdrVersion(fieldName: String, cdrVersion: CdrVersion): List<DataDictionaryEntry>
}
